plugins/telemetry-otel/src/main/java/org/opensearch/telemetry/tracing/OTelTracingTelemetry.java

```diff
@@ -20,29 +22,27 @@
 /**
  * OTel based Telemetry provider
  */
-public class OTelTracingTelemetry implements TracingTelemetry {
+public class OTelTracingTelemetry<T extends OpenTelemetry & Closeable> implements TracingTelemetry {
 
     private static final Logger logger = LogManager.getLogger(OTelTracingTelemetry.class);
-    private final OpenTelemetry openTelemetry;
+    private final T openTelemetry;
+    private final Closeable tracerProviderClosable;
     private final io.opentelemetry.api.trace.Tracer otelTracer;
 
     /**
-     * Creates OTel based Telemetry
+     * Creates OTel based {@link TracingTelemetry}
      * @param openTelemetry OpenTelemetry instance
+     * @param tracerProviderCloseable closable to close the tracer
      */
-    public OTelTracingTelemetry(OpenTelemetry openTelemetry) {
+    public OTelTracingTelemetry(T openTelemetry, Closeable tracerProviderCloseable) {
```

Comment:
Sorry, the fix for OTelTracingTelemetry is a bit more complicated:
```suggestion
public class OTelTracingTelemetry<T extends SdkTracerProvider & Closeable> implements TracingTelemetry {
    public OTelTracingTelemetry(OpenTelemetry openTelemetry, T tracerProvider) {
      ...
    }
    
    @Override
    public void close() throws IOException {
        tracerProvider.close();
    }
}
```
```

Reply:
Yes, this is what I proposed earlier.

Reply:
My apologies, I misunderstood that